cli/src/schema.ts

```diff
@@ -0,0 +1,46 @@
+import { z } from 'zod';
+
+// We need to do this because of problems with Zod and recursive types https://www.npmjs.com/package/zod#recursive-types
+export type Column = {
+  name: string;
+  type: 'bool' | 'int' | 'float' | 'string' | 'text' | 'email' | 'multiple' | 'link' | 'object';
+  unique?: boolean;
+  description?: string;
+  link?: {
+    table: string;
+  };
+  columns?: Column[];
+};
```

Comment:
We should probably use the types from ``@xata.io/client``, so that tsc errors out when we add new types (like date soon)

Reply:
Yeah, I'm not sure how to do that and use it in the zod schema.

I've tried this but I get a weird error

<img width="925" alt="image" src="https://user-images.githubusercontent.com/50486/171031267-3f70731b-221b-490d-87eb-46e1c47b6f00.png">

In any case, if the types don't match we will get a compilation error because we use the zod-generated schema when calling `xata.branches.getBranchMigrationPlan`. If the API spec schema and the zod schema are not compatible, we will know. However it'd be ideal to have the types from the API spec directly, I agree.